korgan/production_cost_speed_optimizer.py

```
@@ -0,0 +1,556 @@
+        successful = 0
+        failures: list[str] = []
+
+        for act_id in sorted(KNOWN_ACTS):
+            staged = target.with_name(f"{target.name}.{act_id}.verified")
+            staged.unlink(missing_ok=True)
+            try:
+                with LegalCorpus(staged) as staged_corpus:
+                    loaded, source_kind, source_url = loader(staged_corpus, act_id)
+                merged = _merge_staged_act(target, staged, act_id)
+                if loaded <= 0 or merged <= 0:
+                    raise RuntimeError(f"empty verified act {act_id}")
```

Ruff (0.16.2)

[warning] Abstract `raise` to an inner function

(TRY301)